edx/edx-enterprise
# ENT-1286 Unlink inactive SAP learners from the related enterprises

**Description:** As a Enterprise Admin, I need to ensure that employees who have left our company are no longer able to take advantage of our edX benefits. Unlink inactive enterprise learners of SAP Success Factors from all related EnterpriseCustomer(s).

**JIRA:** [ENT-1286](https://openedx.atlassian.net/browse/ENT-1286)

**Dependencies:** N/A

**Merge deadline:** N/A

**Installation instructions:** Install `edx-enterprise` from branch `zub/ENT-1286-unlink-inactive-sap-learners` in `edx-platform`.

**Testing instructions:**

1. Create an active enterprise ENT with a catalog.
2. Create integrated channel for SAP Success Factors and add it as an IDP for the enterprise ENT
3. Register user A and user B from SAP Success Factors for a course from enterprise ENT
4. Check that both user A and user B are linked with the enterprise ENT
5. Now mark user B in SAP Success Factors as inactive user.
3. Run `unlink_sap_learners` command, .e.g. `./manage.py lms --settings=aws unlink_sap_learners`
4. Verify that now only user A is linked with enterprise ENT and user B is removed from the enterprise ENT.

**Merge checklist:**

  **DO NOT** just add dependencies to `requirements/*.txt` files.

**Post merge:**

**Author concerns:** List any concerns about this PR - inelegant 
solutions, hacks, quick-and-dirty implementations, concerns about 
migrations, etc.
